Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2198 @@
+        {
+            if (setUpAtRuntime)
+            {
+                UpdateCollection();
+            }
+
+            if (useOnPreRender && clipBox != null)
+            {
+                clipBox.UseOnPreRender = true;
+
+                //Subscribe to the preRender callback on the main camera so we can intercept it and make sure we catch
+                //any dynamically created children in our list
+                cameraMethods = CameraCache.Main.gameObject.EnsureComponent<CameraEventRouter>();
+                cameraMethods.OnCameraPreRender += OnCameraPreRender;
+            }
+        }
+
+        private void Update()
+        {
+            //early bail, the component isn't set up properly
+            if (scrollContainer == null) { return; }
+
+            bool nodeLengthCheck = NodeList.Count > (ViewableArea * Tiers);
+
+            // Force the position if the total number of items in the list is less than the scrollable area
+            if (!nodeLengthCheck)
+            {
+                workingScrollerPos = Vector3.zero;
+                ApplyPosition(workingScrollerPos);
+            }
+
+            //recalcualte every frame to prevent any weirdness from moving the scrolling list.
+            thresholdPoint = transform.TransformPoint(finalOffset);
+
+            //The scroller has detected input and has a valid pointer
+            if (isEngaged)
+            {
+                handDelta = initialHandPos - currentPointerPos;
+
+                //Lets see if this is gonna be a click or a drag
+                //Check the scroller's length state to prevent resetting calculation
+                if (!isDragging && nodeLengthCheck)
+                {
+                    //grab the delta value we care about
+                    absAxisHandDelta = (scrollDirection == ScrollDirectionType.UpAndDown) ? Mathf.Abs(handDelta.y) : Mathf.Abs(handDelta.x);
+
+                    //Catch an intentional finger in scroller to stop momentum, this isn't a drag its definitely a stop
+                    if (absAxisHandDelta > (handDeltaMagThreshold * 0.1f) || TimeTest(initialPressTime, Time.time, dragTimeThreshold))
+                    {
+                        scrollVelocity = 0.0f;
+                        avgVelocity = 0.0f;
+
+                        isDragging = true;
+                        velocityState = VelocityState.None;
+
+                        //reset initialHandPos to prevent the scroller from jumping
+                        initialScrollerPos = scrollContainer.transform.localPosition;
+                        initialHandPos = currentPointerPos;
+                    }
+                }
+
+                //Make sure we're actually (near) touched and not a pointer event, do a dot product check            
+                bool scrollRelease = UseNearScrollBoundary ? DetectScrollRelease(transform.forward * -1.0f, thresholdPoint, currentPointerPos, clippingObject.transform, transform.worldToLocalMatrix, scrollDirection)
```

Comment:
It's a little strange that you aren't using the touch input dispatch system (e.g. by adding NearInteractionTouchable and handing touch events). This code seems to be doing its own hit testing. Seems like it's okay for now since it's experimental. But I'm curious, what didn't work when you tried using NearInteractionTouchable?

Reply:
It actually does, what I'm doing is once the ScrollingObjectCollection has deemed an interaction a "scroll", it cuts off the `eventData` to the child item with `Use()`, and then determines when the scroll is released here.

This actually the magic of the list, I don't care _where_ your hand is when scrolling, only that you're on the correct side of the touch plane.